WcaOnRails/app/models/competition.rb

```diff
@@ -194,6 +194,12 @@ def persisted_events_id
     end
   end
 
+  def persisted_uses_qualification?
```

Comment:
This is sort of gross, I'm open to suggestions to make this better.  Without it, one of the tests in competition_spec.rb fails (when changing the id of a competition, line 492).  This is because:

- :uses_qualification? is called in line 185
- which triggers a reload of the competition_events, which is an empty list after changing the ID
- which makes the validation of :main_event_id fail

Reply:
I have no strong feelings about the `persisted_uses_qualification` hack, as this seems to be a general shortcoming with other properties in the `Competition` model as well. Most notably, the failing `main_event_id` validation that you mention in your comment here also relies on the `with_old_id` mechanism itself.

One thing that makes me wonder however: You are not changing anything about the link between competitions and events itself, you are "only" attaching additional data to the event model. Why does this suddenly break an existing test that used to work before?